bc_obps/registration/service/handle_exception.py

```diff
@@ -0,0 +1,16 @@
+from registration.utils import (
+    generate_useful_error,
+)
+from django.core.exceptions import ValidationError
+
+from django.http import Http404
+
+
+def handle_exception(error):
+    if isinstance(error, ValidationError):
+        return 400, {"message": generate_useful_error(error)}
```

Comment:
We could use the [`422 unprocessable entity`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) code here to discriminate from the generic `400 bad request`

Reply:
@Sepehr-Sobhani , do you know if django ninja expects a 400 for validation errors? When I changed it to 422 I got this error: `ninja.errors.ConfigError: Schema for status 422 is not set in response dict_keys([200, 416, 418, 451, 425, 429, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412])`. I'm using the `from ninja.responses import codes_4xx` so I'm wondering why that wouldn't include 422 if we were meant to be using it

Reply:
You are correct. HTTP status code 422 is reserved and triggers errors when the pydantic schema fails to process the data. So, If there's an issue with the schema we don't even get to this part of the code to raise an error and django-ninja automatically returns the 422 error for us.
A related topic(somehow :) ): https://github.com/vitalik/django-ninja/issues/491#issuecomment-1170990866

Reply:
Wow, really? I was totally just guessing haha

Reply:
Told you, you nailed Python!

Reply:
Still, I think we should add `422` manually since it's the most suitable code for informative errors.